### PR TITLE
ardupilot: added MAV_CMD_EXTERNAL_POSITION_ESTIMATE

### PR DESCRIPTION
**Pull request by tridge**
for handling an external position estimate while dead reckoning

### CODE REVIEW

**Comment by peterbarker** [comment]
Why a command?

Statements of fact aren't usually commands.  What's the receiving system going to do *apart* from ACCEPT the thing?

**Comment by peterbarker** [comment]
.... we already have many, many methods of getting external positions into the autopilot.  This is different to all of them.

I think I see the use case is different here - this is an operator command from a GCS which deserves acknowledgement.  It was the name that threw me - maybe `MAV_CMD_SET_OPERATOR_ESTIMATE_OF_POSITION` would be clearer.

**Comment by rmackay9** [comment]
I've got a similar command that I wanted to add to allow a user to correct a vision position estimate back to a known position.  I just completed testing of this on Saturday using the ModalAI camera. https://github.com/rmackay9/mavlink/commits/set-local-ned-offset

I will raise a PR shortly.

File: message_definitions/v1.0/ardupilotmega.xml [hunk 1]
```diff
@@ -333,6 +333,16 @@
         <param index="6">Empty</param>
         <param index="7">Empty</param>
       </entry>
+      <entry value="43003" name="MAV_CMD_EXTERNAL_POSITION_ESTIMATE" hasLocation="true" isDestination="false">
```
Comment (tridge) [inline message_definitions/v1.0/ardupilotmega.xml:336]: MAV_CMD_EXTERNAL_GLOBAL_POSITION_ESTIMATE

**Comment by rmackay9** [comment]
It would be good to put this into common.xml and push to upstream.

**Comment by tridge** [comment]
> It would be good to put this into common.xml and push to upstream.

why does it need to be in common to go upstream?

**Review by rmackay9 — APPROVED** [review]

**Comment by tridge** [comment]
image

**Comment by hamishwillee** [comment]
>> It would be good to put this into common.xml and push to upstream.

> why does it need to be in common to go upstream?

Does not. As it is in ardupilotmega.xml, upstream will accept a PR commit from this repo without comment provided this is in an ardupilot range.

I suspect it might be more generally useful across flight stacks, but whether I could get commitment to implement from others, as a precondition for being "standard" is another question.

**Pull request merged**